src/main/java/com/desiato/puresynth/controllers/LoginController.java

```diff
@@ -0,0 +1,32 @@
+package com.desiato.puresynth.controllers;
+
+import com.desiato.puresynth.dtos.*;
+import com.desiato.puresynth.services.AuthenticationService;
+import com.desiato.puresynth.services.UserService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/login")
+public class LoginController {
+
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    public LoginController(UserService userService, AuthenticationService authenticationService) {
+        this.userService = userService;
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping
+    public ResponseEntity<LoginResponseDTO> authenticateUser(@RequestBody AuthenticationRequestDTO requestDTO) throws AuthenticationException {
+        PureSynthToken pureSynthToken = authenticationService.authenticate(requestDTO)
+                .orElseThrow(() -> new BadCredentialsException("Unauthorized: Invalid credentials"));
```

Comment:
It would look better, if the `authenticationService.authenticate(requestDTO)` threw the `BadCredentialsException` inside. 

The idea is, that the service layer should contain 95% of the logic. In any other situation, you won't have anything useful to do, if the returned token is `Optional.empty()`. Almost always you need to unpack it, and then throw the exception.

Reply:
Also why are you throwing another exception, than `AuthenticationException`? What value do you get from having both `AuthenticationException` and `BadCredentialsException` in your code? For me it looks like unnecessary complexity. You need to handle 2 different exceptions, you need to cover code with 2 different exception handlers.

Reply:
I tried initially to use only ```AuthenticateException```, but I discovered that, ```AuthenticationException``` is an abstract class, which requires a concrete subclass or an anonymous class implementation when thrown. That's why I used the ```BadCredentialsException```.
Now I have used ```AuthenticateException``` without implementation.

Please let me know if this is ok.